compiler/rustc_target/src/spec/base/apple/mod.rs

```diff
@@ -309,29 +309,10 @@ fn link_env_remove(os: &'static str) -> StaticCow<[StaticCow<str>]> {
     // that's only applicable to cross-OS compilation. Always leave anything for the
     // host OS alone though.
     if os == "macos" {
-        let mut env_remove = Vec::with_capacity(2);
-        // Remove the `SDKROOT` environment variable if it's clearly set for the wrong platform, which
```

Comment:
Is `rustc` now relying on `cc` or `clang` to sort this out since `SDKROOT` is unconditionally passed now?

Reply:
Before, this code was only really useful when linking with `cc` on macOS (because in other cases, we passed the `-isysroot` parameter on the commandline, which overwrote it).

Now, we explicitly set the `SDKROOT` environment variable when linking with `cc`, so it'll get overwritten.

Reply:
Oh, you meant the logic itself.

That was somewhat duplicated between this function and `get_apple_sdk_root`, I've made sure to merge it properly now.

Reply:
Aaaa, nice :)